Plugins/ExaTrkX/src/Tensor.cpp

```diff
@@ -0,0 +1,166 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/ExaTrkX/Tensor.hpp"
+
+#ifdef ACTS_EXATRKX_WITH_CUDA
+#include "Acts/Plugins/ExaTrkX/detail/CudaUtils.hpp"
+#endif
+
+#include <cstring>
+#include <numeric>
+
+namespace Acts {
+
+namespace detail {
+
+TensorMemoryImpl::TensorMemoryImpl(std::size_t nbytes,
+                                   const ExecutionContext &execContext)
+    : m_device(execContext.device) {
+  if (execContext.device.type == Acts::Device::Type::eCPU) {
+    m_ptr = std::malloc(nbytes);
```

Comment:
Why not use `new` / `delete`?

Reply:
hmm `new void[size]` seems not to work...

Reply:
Best is `new std::byte[size]`.

Reply:
Hmm so when I use `std::byte *` as type of the member, I get a warning of the compiler when converting `std::byte **` to `void **` for the CUDA API functions... Is there a way how to do this ideomatic with the CUDA API?
`std::malloc`/`std::free` is the only way I saw without any compiler warnigns so far...

Reply:
Which compiler is that?
As far as I can tell, clang and gcc are happy with:

```cpp
#include <cstddef>

struct S {
    void* m_ptr;
};

int main() {
    S s;
    s.m_ptr = new std::byte[10];

    void** p = &s.m_ptr;
}
```

Btw: In principle you could also use `std::unique_ptr` with a custom deleter. That way, you would not have to specify the destructor yourself.

Reply:
So the issue is not standard C++ but CUDA: I would need
```
std::byte *ptr;
cudaMalloc(&ptr, ...); // -> this gives a warning about converting `std::byte **` to `void**`
```
Only workaround is a C-style cast I think like `(void **)ptr`, or maybe `std::reinterpret_cast`. Not sure if this is nicer than using void pointer?

Reply:
Hmm okay I see. so we store void ptr, but allocate `std::byte`...

Reply:
Okay, implemented.